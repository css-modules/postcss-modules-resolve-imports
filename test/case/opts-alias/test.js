'use strict';

const {resolve} = require('path');
const resolveImports = require('../../../index');
const setup = require('../../setup');

test('opts-alias', () => {
  const {resulting, exports: tokens} = setup(
    'local-by-default',
    'extract-imports',
    'scope',
    resolveImports({
      resolve: {
        alias: {
          button: resolve(__dirname, 'lib/button/button.css'),
        },
      },
    })
  )(__dirname);

  expect(resulting).toMatchSnapshot();
  expect(tokens).toMatchSnapshot();
});
