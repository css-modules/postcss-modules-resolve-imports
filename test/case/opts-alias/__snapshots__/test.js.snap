// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opts-alias 1`] = `
"/* imported from lib/button/button.css */

._button_button
{
  /*common button styles*/
}

/* imported from source.css */

._source_continueButton
{
  color: green;
}
"
`;

exports[`opts-alias 2`] = `
Object {
  "continueButton": "_source_continueButton _button_button",
}
`;
